### DISCUSSION TITLE
migrate from DB-less to postgres DB

### DISCUSSION THREAD

**Discussion by balait4** [Kong/kong#13304] [Help] [answered]
Hi All,

Using kong 3.5.0 standalone installation (DB-less) in openshift cluster and like to know below 2 request please:
*) Is there any documentation how to migrate from DB-less to Postgres DB?
*) We are planning to extend our plugin with custom Admin API, So change the configuration of a plugin via the admin API will not be persisted with DB-less mode? So is there a way to do this on DB-less mode? 

Thanks

**Comment by hanshuebner** [comment, marked as answer]
> Is there any documentation how to migrate from DB-less to Postgres DB?

To my knowledge, there is no such document.  Importing a declarative configuration file into Kong Gateway's database is possible using the [`kong config db_import`](https://docs.konghq.com/gateway/latest/reference/cli/#kong-config) command, which may be your best route.

> We are planning to extend our plugin with custom Admin API, So change the configuration of a plugin via the admin API will not be persisted with DB-less mode? So is there a way to do this on DB-less mode?

I'm not sure if I understand the question quite well.  In dbless mode, your configuration is read from a file when Kong Gateway starts.  Changes that are made to the configuration at run time are not persisted, and Kong Gateway does not update the declarative configuration file that was read at startup time

**Comment by balait4** [comment]
Thanks for the reply, we are running in openshift cluster where declarative configuration stores in ETCD. So we are thinking if its possible to extend our plugin with custom Admin API like here: https://docs.konghq.com/gateway/latest/plugin-development/admin-api/

**Reply by hanshuebner** [reply]
What do you want to achieve by extending the Admin API?

**Reply by balait4** [reply]
Its mainly to manage custom plugins via Admin API that aren't natively supported by Kong.

**Reply by hanshuebner** [reply]
If you need persistent state (i.e. DAO objects) in your plugin's Admin API extension, then you cannot use dbless mode.

**Reply by balait4** [reply]
ok thanks, thats the reason I was check to migrate from DB-less to postgresDB
